website/integrations/services/argocd/index.md

```diff
@@ -0,0 +1,105 @@
+---
+title: ArgoCD
+---
+
+<span class="badge badge--secondary">Support level: Community</span>
+
+## What is ArgoCD
+
+From https://argoproj.github.io/cd/
+
+:::note
+Argo CD is a declarative, GitOps continuous delivery tool for Kubernetes.
+:::
+
+## Preparation
+
+The following placeholders will be used:
+
+-   `argocd.company` is the FQDN of the ArgoCD install.
+-   `authentik.company` is the FQDN of the authentik install.
+
+:::note
+In this guide, only modified values are appearing , others are left at their default values.
+:::
+
+## Authentik Configuration
+
+### Step 1 - Provider creation
+
+In authentik, create an _OAuth2/OpenID Provider_ (under _Applications/Providers_) with these settings:
+
+-   Name: ArgoCD
+-   Signing Key: Select any available key
+-   Redirect URIs:
+
+```
+http://argocd.company/api/dex/callback
+http://localhost:8085/auth/callback
+```
+
+After creating the provider, take note of the `Client ID` and `Client Secret`, you'll need to give them to ArgoCD in the _ArgoCD Configuration_ field.
+
+### Step 2 - Application creation
+
+Create a new _Application_ (under _Applications/Applications_) with these settings:
+
+-   Name: ArgoCD
+-   Provider: ArgoCD
+
+### Step 3 - ArgoCD Admin Group creation
+
+Create a new _Group_ (under _Directory/Groups_) that'll be used as the admin group for ArgoCD (if you already have an "admin" group, you can skip this part!)
+
+-   Name: ArgoCD Admins
+-   Members: Add your user and/or any user that should be an ArgoCD admin
+
+## ArgoCD Configuration
+
+:::note
+We're not going to use the oidc config, but instead the "dex", oidc doesn't allow ArgoCD CLI usage while DEX does.
```

Comment:
If possible link some docs as a source for this

Reply:
I'll try to find back the different issues/docs that documented this!

Reply:
Found back the issue, it wasn't exactly about OIDC itself, for Google the switch to DEX was necessary to have both the web & cli to work, and it's the only setting that worked for me to get the CLI working (I wasn't able to login though it without switching over to DEX instead of OIDC : https://github.com/argoproj/argo-cd/issues/7940)

Reply:
After checking a bit more Argo's doc, it seems specifying the `cliClientId` field is sometimes necessary : https://argo-cd.readthedocs.io/en/stable/operator-manual/user-management/#existing-oidc-provider however, that shouldn't be needed *in theory* as the field seems automatically "set" with the value of the (mandatory) `clientId` field

I'll give it a try on my own ArgoCD and update accordingly :)